tools/scripts/monitoring/structure-health-check.sh

```
@@ -0,0 +1,228 @@
+  # Check for oversized directories
+  echo ""
+  echo "📊 Top 10 directory sizes:"
+  du -sh */ 2>/dev/null | sort -hr | head -10 | while read size dir; do
+    echo "   $size $dir"
+  done
+
+  # Check for duplicate TypeScript files
+  echo ""
+  echo "🔍 Checking for duplicate TypeScript file names..."
+  duplicates_found=false
+  find . -type f -name "*.ts" -not -path "*/node_modules/*" | \
```

Shellcheck (0.10.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)